app/helpers/shf_icons_helper.rb

```diff
@@ -1,32 +1,263 @@
+  #   For this entry,  method_name_start: 'edit'  so the methods defined will start with "edit"
+  #   The full method names will be "edit_icon" and "edit_fa_icon_name"
+  #
+  #   In the entry, icon: FA_EDIT  means that the FontAwesome icon defined by
+  #   FA_EDIT will be used. Since FA_EDIT = 'edit', the the FontAwesome
+  #   icon used will be "fa-" + "edit" = "fa-edit"
+  #
+  #   These are the 2 methods defined that can be used in views and controllers
+  #   as helper methods:
+  #
+  #      def edit_icon(html_options: {}, fa_style: FA_STYLE_DEFAULT, text: nil)
+  #         get_fa_icon(fa_style, FA_EDIT, text, html_options)
+  #       end
+  #
+  #      def edit_fa_icon_name
+  #        "fa-edit"
+  #      end
+  #
+  # You can then use those helper methods in views:
+  #   = link_to edit_icon, admin_only_user_profile_edit_path(user), title: t('.edit_user_profile', name: user.full_name)
+  #
+  # instead of the old way where you had to specify the FontAwesome icon to use:
+  #   = link_to icon('fas', 'edit'), admin_only_user_profile_edit_path(user), title: t('.edit_user_profile', name: user.full_name)
+  #
+  #
+  # @example
+  #  In a view, instead of writing this:
+  #
+  #  # OLD WAY (BAD)
+  #  - trash_icon = icon('fas', 'trash')
+  #  = link_to trash_icon, remove_attachment_shf_application_path(shf_application.id,
+  #                             shf_application: { uploaded_files_attributes: { id: uploaded_file.id,  '_destroy' => true }}),
+  #                           method: :put, remote: true,
+  #                           id: "uploaded-file-#{uploaded_file.id}",
+  #                           class: "action-delete",
+  #                           data: { confirm: "#{t('shf_applications.uploads.confirm_delete', filename: uploaded_file.actual_file_file_name)}" }
+  #
+  #
+  # use the helper method like this:
+  #
+  # # NEW WAY WITH THE HELPER METHODS (GOOD)
+  # = link_to destroy_icon, remove_attachment_shf_application_path(shf_application.id,
+  #                             shf_application: { uploaded_files_attributes: { id: uploaded_file.id,  '_destroy' => true }}),
+  #                           method: :put, remote: true,
+  #                           id: "uploaded-file-#{uploaded_file.id}",
+  #                           class: "action-delete",
+  #                           data: { confirm: "#{t('shf_applications.uploads.confirm_delete', filename: uploaded_file.actual_file_file_name)}" }
+  #
+  #
+  #  @example:
+  #   assuming FA_COMPLETE_CHECK = 'check-circle'
+  #   for this entry:  { method_name_start: 'complete_check', icon: FA_COMPLETE_CHECK },
+  #
+  #   these methods are defined:
+  #
+  #      def complete_check_icon(html_options: {}, fa_style: FA_STYLE_DEFAULT, text: nil)
+  #         get_fa_icon(fa_style, FA_COMPLETE_CHECK, text, html_options)
+  #       end
+  #
+  #      def complete_check_fa_icon_name
+  #        "fa-check-circle"
+  #      end
+  #
+  #
```

Comment:
This is great.  Although you have plenty of documentation, it might be nice to have an example of an icon with a tooltip (a useful thing IMO since not everyone will necessarily be familiar with a specific icon _or_ the action of the icon may not be clear from the context).  

Something like
```
destroy_icon(html_options: { 'data-toggle': 'tooltip', title: t('addresses.edit.delete') })
```

Reply:
@patmbolger  - I added some examples.  Kindly proofread because I wrote this without enough coffee in my system.